api/v1beta1/hetznerbaremetalhost_webhook.go

```diff
@@ -40,19 +55,41 @@ func (host *HetznerBareMetalHost) Default() {
 
 //+kubebuilder:webhook:path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-hetznerbaremetalhost,mutating=false,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=hetznerbaremetalhosts,verbs=create;update,versions=v1beta1,name=validation.hetznerbaremetalhost.infrastructure.cluster.x-k8s.io,admissionReviewVersions={v1,v1beta1}
 
-var _ webhook.Validator = &HetznerBareMetalHost{}
+var _ webhook.CustomValidator = &HetznerBareMetalHostWebhook{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (host *HetznerBareMetalHost) ValidateCreate() (admission.Warnings, error) {
-	return nil, nil
+func (host *HetznerBareMetalHostWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	hetznerBareMetalHostObj, ok := (obj).(*HetznerBareMetalHost)
+	if !ok {
+		return admission.Warnings{}, errors.NewBadRequest(fmt.Sprintf("expected HetznerBareMetalHost, but got %T", hetznerBareMetalHostObj))
+	}
+
+	var allErrs field.ErrorList
+
+	hetznerBareMetalHostList := &HetznerBareMetalHostList{}
+	if err := host.client.List(ctx, hetznerBareMetalHostList, &ctrlClient.ListOptions{Namespace: hetznerBareMetalHostObj.Namespace}); err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(field.NewPath("spec", "ServerID"), fmt.Errorf("could not verify that the host has a unique serverID")),
+		)
+	}
+
+	for _, hetznerBareMetalHost := range hetznerBareMetalHostList.Items {
+		if hetznerBareMetalHost.Spec.ServerID == hetznerBareMetalHostObj.Spec.ServerID {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec", "ServerID"), hetznerBareMetalHostObj.Spec.ServerID, fmt.Sprintf("other host exist with same ServerID: %d", hetznerBareMetalHostObj.Spec.ServerID)),
```

Comment:
maybe mention not the serverID (that's obvious anyway), maybe mention the name of the other host object with the same ServerID

Reply:
```go
return admission.Warnings{"could not verify that the host has a unique serverID"}, nil
```

Lint failure - `error is not nil (line 70) but it returns nil (nilerr)`